pkg/policy/selectorcache_selector.go

```diff
@@ -182,25 +182,13 @@ func (f *fqdnSelector) setSelectorIPs(ips []netip.Addr) {
 // that is in wantLabels.
 // This is reasonably efficient, as it relies on both arrays being sorted.
 func (f *fqdnSelector) matches(identity scIdentity) bool {
-	wantIdx := 0
-	checkIdx := 0
-
-	// Both arrays are sorted; walk through until we get a match
-	for wantIdx < len(f.wantLabels) && checkIdx < len(identity.lbls) {
-		want := f.wantLabels[wantIdx]
-		check := identity.lbls[checkIdx]
-		if want == check {
-			return true
-		}
-
-		// Not equal, bump
-		if check.Key < want.Key {
-			checkIdx++
-		} else {
-			wantIdx++
+	for _, wantLbl := range f.wantLabels {
+		for _, haveLbl := range identity.lbls {
+			if wantLbl.Matches(&haveLbl) {
```

Comment:
nit, Mildly curious whether there's anything to look into here in that until now `Label` didn't have a public `Matches` but we always used `LabelArray` and its corresponding `Has` / `Contains` / `Lacks` APIs. Not a big deal but maybe rather than iterating `identity.lbls` it could/should just use `identity.Has(...)`...?

Reply:
I added a new method, `Intersects()`, that does the same thing. Perhaps it's useful. Otherwise, yes, changing this to `.Has()` would be the right idea.